### PR TITLE
Add workflow pipelines

### PR DESCRIPTION
**Pull request by ausias-armesto**
Add default pipelines to repository

## Summary by CodeRabbit

- **New Features**
	- Introduced automated workflows for building, merging, and releasing the application.
	- Added support for multiple Node.js versions in the build process.
	- Implemented a manual trigger for closing software releases with versioning options.
	- Enhanced release notifications to a Zulip stream.
	- Added security measures by ignoring sensitive credential files in version control.

- **Bug Fixes**
	- Removed outdated CI workflow for improved efficiency and clarity.

### CODE REVIEW

**Review by esterlus — CHANGES_REQUESTED** [review]
Can you update README.md with description of new CHANGELOG handling and deployments?

File: README.md [hunk 1]
```diff
@@ -4,14 +4,10 @@
...
-## Deployment
+## Deployment process
 
-CI will create a new container image on a tagged version on main.
+To contribute to this repository you will need to create a pull request. More information about the existing automated workflows can be found in [GitHub Actions](./.github/workflows/README.md)
 
--   run `yarn changeset version` to create the current changelog
--   run `yarn build` to update version info
--   commit everything, create a matching tag and push to main
--   update container tag `stable` in container registry as needed
```
Comment (esterlus) [inline README.md:14]: We wil still manually add a stable tag, right?

**Comment by esterlus** [comment]
Looks good, one last thing:

Does it tag the container with `stable` during creating a release?

**Comment by ausias-armesto** [comment]
> Can you update README.md with description of new CHANGELOG handling and deployments?

Changelog in repositories like this one and uhttp-latency-monitor where the artifact is a docker image instead of a npm package, the changelog generation is different.
The "close release" workflow is in charge of creating the Github release, but as I'm not providing any content into the release, is it taking the commit linear history of the release. As we have forced to use PR to contribute to main, if we squash all our PR then the changelog will contain the list of PR.
This can be improved but requires more bureaucracy that we don't want at this moment.

**Comment by ausias-armesto** [comment]
> r



> Looks good, one last thing:
> 
> Does it tag the container with `stable` during creating a release?

No, I've not included that feature yet, as we might want to test the version in some other environments before mark it as `stable`. Something to consider for future improvements.

**Comment by esterlus** [comment]
What happens with the existing changelog then?

**Review by esterlus — APPROVED** [review]
Let's try the new workflow then

**Pull request merged**